### PR TITLE
Stop the robots when the stop receiving packages for 1 sec

### LINKED ISSUE #166: Make the robots stop moving after the simulator stopped receiving packets to move the robots for some time
**Is your feature request related to a problem? Please describe.**
Yes, when someone's AI is turned off, the robots keep moving, which causes some confusion on the field. They should be halted in those situations.

**Describe the solution you'd like**
The robots could stop moving if they do not receive a package to do so for 100 ms, 500 ms or something like that.

**Describe alternatives you've considered**
Nothing yet.

**Additional context**
Nothing.

### PR DESCRIPTION
**Pull request by Bollos00**
### Identify the Bug

Closes #166 

### Description of the Change

* Whenever a socket is received from `blueControlSocket`, the `QElapsedTimer` `elapsedLastPackageBlue` is restarted.
* The function `MainWindow::update` will check every time it is called if `elapsedLastPackageBlue` has not been restarted for more than one second. If the condition is true, it will stop all blue robot. A similar/equivalent approach was made for the `yellowControlSocket`.

### Alternate Designs

Nothing.

### Possible Drawbacks

* It does not turn the roller and the kick (I still did not find the proper way of doing so);
* I am still testing it.

### Release Notes

Stop the robots when the stop receiving packages for 1 second

### CODE REVIEW

**Comment by Bollos00** [comment]
@mahi97 I tested it with a match test and I think this pull request is ready for review now.

One question, should I change something on the piece of code complained bu the CodeFactor?

**Comment by Bollos00** [comment]
@g3force this pull request is open for quite some time now... do I need to change anything before the pull request is merged?

File: src/mainwindow.cpp [hunk 1]
```diff
@@ -410,6 +410,42 @@ void MainWindow::update()
...
+            for(int i=0; i < glwidget->cfg->Robots_Count(); ++i)
+            {
+                const int index = glwidget->ssl->robotIndex(i, BLUE-1);
+
+                if(index == -1 ||
+                   glwidget->ssl->robots[index] == nullptr)
+                    continue;
+
+                glwidget->ssl->robots[index]->resetSpeeds();
+//                glwidget->ssl->robots[index]->kicker->setRoller(0);
```
Comment (g3force) [inline src/mainwindow.cpp:428]: Why is this commented out?

File: src/mainwindow.cpp [hunk 2]
```diff
@@ -410,6 +410,42 @@ void MainWindow::update()
...
+        // Stops blue robots from moving if no package has been received for 1 second
+        if(glwidget->ssl->elapsedLastPackageBlue.nsecsElapsed()*1e-9 > 1)
+        {
+            for(int i=0; i < glwidget->cfg->Robots_Count(); ++i)
+            {
+                const int index = glwidget->ssl->robotIndex(i, BLUE-1);
+
+                if(index == -1 ||
+                   glwidget->ssl->robots[index] == nullptr)
+                    continue;
```
Comment (g3force) [inline src/mainwindow.cpp:425]: Some `{}` would be nice here, as the `if` is already on two lines

**Review by g3force — CHANGES_REQUESTED** [review]
Looks good to me except for the comments.

File: src/mainwindow.cpp (hunk 1 above)
Comment (Bollos00) [reply src/mainwindow.cpp:428]: I was testing and forgot to remove it.

**Comment by Bollos00** [comment]
@g3force what about now?

**Comment by g3force** [comment]
I've fixed the pipelines on master und updated your branch to see if everything is green.

Also, please use the GitHub review functions:
image

Press the cyclic arrows to rerequest review.

**Comment by g3force** [comment]
And please fix the CodeFactor issue. One method you modified is too complex now. Move code into a new method.

**Review by g3force — APPROVED** [review]

**Pull request merged**